platforms/software/build-init/src/main/java/org/gradle/api/tasks/wrapper/Wrapper.java

```diff
@@ -329,9 +330,13 @@ public void setDistributionType(DistributionType distributionType) {
     /**
      * The list of available gradle distribution types.
      */
-    @ToBeReplacedByLazyProperty(comment = "Not supported yet", issue = "https://github.com/gradle/gradle/issues/29341")
+    @Deprecated
     @OptionValues("distribution-type")
     public List<DistributionType> getAvailableDistributionTypes() {
+        DeprecationLogger.deprecateMethod(Wrapper.class, "getAvailableDistributionTypes")
+            .willBeRemovedInGradle10()
+            .withUpgradeGuideSection(9, "deprecated_wrapper_get_available_distribution_types")
+            .nagUser();
         return Arrays.asList(DistributionType.values());
```

Comment:
❓ Do we have a test that verifies the equivalence of `DistributionType.values()` and the result of this method? If we're to guarantee it, I'd rather have this explicitly tested. It can be a unit test.

Reply:
Added a test